src/engine/Join.h

```diff
@@ -94,6 +94,18 @@ class Join : public Operation {
   void join(const IdTable& a, ColumnIndex jc1, const IdTable& b,
             ColumnIndex jc2, IdTable* result) const;
 
+ private:
+  // Part of the implementation of `createResult`. This function is called when
+  // the result should be yielded lazily.
+  // The semantics of action can be seen as
+  // runJoinAndReturnFinalResult(callbackForIntermediateResults).
+  Result::Generator yieldOnCallbackCalled(
```

Comment:
```suggestion
  Result::Generator runLazyJoinAndConvertToGenerator(
```
(thats'what it is, and the argument is then called `runJoin`. or `runLazyJoin`.